src/cluster.c

```diff
@@ -1999,13 +1999,28 @@ int writeHostnamePingExt(clusterMsgPingExt **cursor) {
     uint32_t extension_size = getHostnamePingExtSize();
 
     /* Move the write cursor */
-    (*cursor)->type = CLUSTERMSG_EXT_TYPE_HOSTNAME;
+    (*cursor)->type = htons(CLUSTERMSG_EXT_TYPE_HOSTNAME);
     (*cursor)->length = htonl(extension_size);
     /* Make sure the string is NULL terminated by adding 1 */
     *cursor = (clusterMsgPingExt *) (ext->hostname + EIGHT_BYTE_ALIGN(sdslen(myself->hostname) + 1));
     return extension_size;
 }
 
+/* Write the forgotten node ping extension at the start of the cursor, update
+ * the cursor to point to the end of the written extension and return the number
+ * of bytes written. */
+int writeForgottenNodePingExt(clusterMsgPingExt **cursor, sds name, uint64_t ttl) {
+    serverAssert(sdslen(name) == CLUSTER_NAMELEN);
+    clusterMsgPingExtForgottenNode *ext = &(*cursor)->ext[0].forgotten_node;
+    memcpy(ext->name, name, CLUSTER_NAMELEN);
+    ext->ttl = htonu64(ttl);
+    uint32_t extension_size = sizeof(clusterMsgPingExt) + sizeof(clusterMsgPingExtForgottenNode);
+    (*cursor)->type = htons(CLUSTERMSG_EXT_TYPE_FORGOTTEN_NODE);
+    (*cursor)->length = htonl(extension_size);
+    *cursor = (clusterMsgPingExt *) (ext->name + sizeof(clusterMsgPingExtForgottenNode));
```

Comment:
is clusterMsgPingExtForgottenNode should be aligned with 8 bytes?

Reply:
The size of clusterMsgPingExtForgottenNode is a multiple of 8, so it is already aligned. But perhaps I should add an assert?

Reply:
I added `static_assert(sizeof(clusterMsgPingExtForgottenNode) % 8 == 0);`

Reply:
Seems like there is a `UndefinedBehaviorSanitizer`, i ran a daily and it report this
```
cluster.c:2053:48: runtime error: index 48 out of bounds for type 'char [40]'
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior cluster.c:2053:48 in 
```